### PR TITLE
MANT: handling of axis in np.ma.compress_nd

### PR DESCRIPTION
**Pull request by jaimefrio**
Modified the handling that np.ma.compress_nd does of its 'axis'
argument, to more closely follow that of ufunc methods, including
error messages for wrong values.

### CODE REVIEW

File: numpy/ma/extras.py [hunk 1]
```diff
@@ -731,29 +731,27 @@ def compress_nd(x, axis=None):
...
-        raise ValueError('Invalid type for axis argument')        
+    elif not isinstance(axis, tuple):
+        raise ValueError('Invalid type for axis argument')
     # Check axis input
-    for ax in axis:
-        if not (0 <= ax < x.ndim):
-            raise ValueError('axis %d is out of range' % ax)
-    if not len(axis) == len(set(axis)):
-        raise ValueError('axis cannot have dupliate entries')
+    axis = [ax + (x.ndim if ax < 0 else 0) for ax in axis]
```
Comment (charris) [inline numpy/ma/extras.py]: Can write this `[ax + x.ndim if ax < 0 else  ax for ax in axis]`.

File: numpy/ma/extras.py [hunk 2]
```diff
@@ -731,29 +731,27 @@ def compress_nd(x, axis=None):
...
+    if any(ax < 0 or ax >= x.ndim for ax in axis):
```
Comment (charris) [inline numpy/ma/extras.py]: Can also write this `if not all(0 <= ax < x.ndim aor ax in axis):`.

**Comment by charris** [comment]
LGTM, just a couple of alternative implementation suggestions.

**Comment by jaimefrio** [comment]
Your wish is my command. ;-)

**Comment by charris** [comment]
Thanks Jaime.

**Pull request merged**